watershed-fire/2_process/src/map_data.R

```diff
@@ -1,9 +1,25 @@
+# Process fire perimeter data
+prep_perims <- function(perim){
+  perim %>%
+    mutate(Year = as.numeric(str_sub(Ig_Date, start = 1L, end = 4L)),
+           State = str_sub(Event_ID, start = 1L, end = 2L)) %>%
+    filter(Incid_Type %in% c("Wildfire", "Wildland Fire Use"),
+           State %in% conus) %>%
+    select(Event_ID, Incid_Name, Ig_Date, Year, State, Incid_Type, BurnBndAc, BurnBndLat, BurnBndLon) %>%
+    st_transform(crs = crs)
+}
+
 # Convert fire perimeters to points
 sf2df <- function(sf, years){
-  st_centroid(sf) %>%
+  out <- sf %>%
+    filter(Year %in% years)
+  out %>%
+    st_centroid() %>%
     st_geometry() %>%
     do.call(rbind, .) %>%
     as.data.frame() %>%
     setNames(c("lon","lat")) %>%
-    bind_cols(st_set_geometry(sf, NULL))
-}
\ No newline at end of file
+    bind_cols(st_set_geometry(out, NULL)) %>%
+    add_row(Year = years + 0.5)
```

Comment:
What is this doing?

Reply:
Line 22 is binding the data from the polygon dataset to the centroid dataset.

Line 23 is add blank rows between years. To produce the embers glowing effect of the map, it has to fade into a blank map (or more specifically, into a map with subtle glowing of all the previous fires' "scars") between the annual glowpoint frames. This adds these "blank frames" to fade in and out of between years. Here is an example of what those frames look like.

2003
![image](https://user-images.githubusercontent.com/25806255/180872847-eadacf7c-ea47-4f20-89e3-bda7064fb165.png)

2003.5
![image](https://user-images.githubusercontent.com/25806255/180872885-9be47ac3-846b-4874-a30b-579769bb98f6.png)

2004
![image](https://user-images.githubusercontent.com/25806255/180873008-c2392fa3-6991-4ee0-a581-1f6661f6cc94.png)

Without these intermediate frames it gets difficult to discern what is happening each year.

Reply:
smart!